strawberryfields/cli/__init__.py

```diff
@@ -0,0 +1,311 @@
+# Copyright 2019-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A standalone command-line interface for configuring Strawberry Fields and connecting
+to the Xanadu cloud platform.
+"""
+
+import argparse
+import sys
+
+from strawberryfields.api import Connection
+from strawberryfields.configuration import (ConfigurationError, create_config,
+                                            store_account)
+from strawberryfields.engine import StarshipEngine
+from strawberryfields.io import load
+
+
+def main():
+    """The Xanadu cloud platform command line interface.
+
+    **Example:**
+
+    The following is a simple example on getting the help message of the cloud platform command line interface. It details each of the options available.
+
+    .. code-block:: bash
+
+        $ starship
+        usage: starship <command> [<args>]
+
+        These are common options when working on the Xanadu cloud platform.
+
+        General Options:
+          -h, --help       show this help message and exit
+          --ping, -p       Tests the connection to the remote backend.
+
+        Commands:
+          {configure,run}
+            configure      Configure each detail of the API connection.
+            run            Run a blackbird script.
+    """
+    parser = create_parser()
+    args = parser.parse_args()
+
+    if args.ping:
+        ping()
+    elif hasattr(args, "func"):
+        args.func(args)
+    else:
+        parser.print_help()
+
+def create_parser():
+    """Creates a parser to process the commands and arguments passed to the
+    command line interface.
+
+    Returns:
+        ArgumentParser: an argument parser object that defines the related
+        options
+    """
+    parser = argparse.ArgumentParser(
+        usage="starship <command> [<args>]",
```

Comment:
That's a great question! I've mostly redefined what is outputted as a help menu, such that it appears nice (but this is entirely personal taste, can change it). Also, in this regard, I sort of tried to follow `pip`'s example:
```shell
$ pip
Usage:   
  pip <command> [options]
```
Specifically, without this, the usage would appear like
```shell
usage: starship [-h] [--ping] {configure,run} ...
```

Which one do you think would make more sense overall? :)

Reply:
I think I actually prefer the second one as it seems a bit more explicit, and not too cluttered since we don't have too many options or commands; but as you point out this is quite subjective, and I think either would be pretty good!

Reply:
Oh, that's great feedback! As I don't have a strong preference here myself, I'll delete the `usage` part for now (and apply your suggestion :slightly_smiling_face: )